upup/pkg/fi/cloudup/populate_instancegroup_spec.go

```diff
@@ -161,6 +161,20 @@ func PopulateInstanceGroupSpec(cluster *kops.Cluster, input *kops.InstanceGroup,
 		return nil, fmt.Errorf("unable to infer any Subnets for InstanceGroup %s ", ig.ObjectMeta.Name)
 	}
 
+	switch kops.CloudProviderID(cluster.Spec.CloudProvider) {
+	case kops.CloudProviderAWS:
+		mt, err := awsup.GetMachineTypeInfo(cloud.(awsup.AWSCloud), ig.Spec.MachineType)
+		if err != nil {
+			return ig, fmt.Errorf("error looking up machine type info: %v", err)
+		}
+		if mt.GPU {
+			if ig.Spec.NodeLabels == nil {
+				ig.Spec.NodeLabels = make(map[string]string)
+			}
+			ig.Spec.NodeLabels["kops.k8s.io/gpu"] = "1"
+			ig.Spec.Taints = append(ig.Spec.Taints, "nvidia.com/gpu:NoSchedule")
```

Comment:
I think this is a behavioral change?  i.e. workloads that previously ran on GPU instances will need to add this toleration to remain schedulable?

Or is there a webhook or similar that will add this automatically?

Reply:
Good catch. I think this one should additionally be blocked by NvidiaGPU.enabled. That way one has explicitly subscribed to kOps way of doing things. I'll add docs to match as well.